pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -775,10 +785,18 @@ func getLocationFromPod(clientset kubernetes.Interface, d *apps.Deployment) (str
 			hostName = pvcName
 		}
 	}
-	return GetLocationWithNode(clientset, nodeName, hostName)
+	// compatibility with existing clusters
+	crushRoot := "default"
+	// find the env var indicating the root crush entry
+	for _, envvar := range pods.Items[0].Spec.Containers[0].Env {
```

Comment:
If someone is running istio, sidecar containers might be injected. You'll need to iterate over the containers to find the container with the expected name.